spotbugs/src/main/java/edu/umd/cs/findbugs/detect/ViewCFG.java

```diff
@@ -0,0 +1,176 @@
+package edu.umd.cs.findbugs.detect;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Iterator;
+
+import org.apache.bcel.classfile.JavaClass;
+import org.apache.bcel.classfile.Method;
+import org.apache.bcel.generic.InstructionHandle;
+
+import edu.umd.cs.findbugs.BugReporter;
+import edu.umd.cs.findbugs.Detector;
+import edu.umd.cs.findbugs.ba.BasicBlock;
+import edu.umd.cs.findbugs.ba.CFG;
+import edu.umd.cs.findbugs.ba.CFGBuilderException;
+import edu.umd.cs.findbugs.ba.ClassContext;
+import edu.umd.cs.findbugs.ba.Edge;
+import static edu.umd.cs.findbugs.ba.EdgeTypes.*;
+
+public class ViewCFG implements Detector {
+
+    private final BugReporter bugReporter;
+    private Path tempDir;
+
+    public ViewCFG(BugReporter bugReporter) {
+        this.bugReporter = bugReporter;
+        try {
+            tempDir = Files.createTempDirectory("cfg-");
+        } catch (IOException e) {
+            bugReporter.logError("Could not create temporary directory", e);
+        }
+    }
+
+    @Override
+    public void visitClassContext(ClassContext classContext) {
+        if (tempDir == null) {
+            return;
+        }
+        //
+        JavaClass cls = classContext.getJavaClass();
+        String classDirName = (cls.getPackageName() != "") ? (cls.getPackageName() + "." + cls.getClassName()) : cls.getClassName();
+        Path classDir;
+        //
+        try {
+            classDir = Files.createDirectory(Paths.get(tempDir.toString(), classDirName));
+        } catch (IOException e) {
+            bugReporter.logError("Could not create directory for class " + cls.getClassName(), e);
+            return;
+        }
+        //
+        for (Method method : cls.getMethods()) {
+            try {
+                analyzeMethod(classContext, method, classDir);
+            } catch (CFGBuilderException e) {
+                bugReporter.logError("Error analyzing method", e);
+            }
+        }
+    }
+
+    private void analyzeMethod(ClassContext classContext, Method method, Path classDir) throws CFGBuilderException {
+        Path methodFile = getMethodFile(classDir, method.getName());
+
+        try (PrintStream out = new PrintStream(Files.createFile(methodFile).toFile(), Charset.defaultCharset().name())) {
+            CFG cfg = classContext.getCFG(method);
+            out.println("digraph " + method.getName() + " {");
```

Comment:
Probably we need to wrap the method name by `"`, to support special names like `<new>`.

Reply:
It is worse. Characters like < and > are not supported on Windows so I changed method names like `<init>` to `____init`.